src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -274,7 +275,7 @@ void primitive_inst::update_impl() {
             return ps.to_shape();
         }
         if (ps.size() < 4) {
-            if (_node->is_type<gemm>())
+            if (_node->is_type<gemm>() || _node->is_type<select>())
```

Comment:
Please check the latest master. extend_to_6d is moved to primivie_base so that it can be overriden.